### PR TITLE
New transaction module

### PR DESCRIPTION
**Pull request by rodrwan**
First version of this refactor.

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -1,71 +1,115 @@
...
 Setters & getters
 -----------------
 Setters and getters are pattern matched functions that behave depending on wether an
 argument was passed or not.
 
-**Description**
-Description cannot be empty
+**Date**
+Date can be passed as a formatted string.
+Onli in format *dd/mm/yyyy* where day and month are allowed to be prefixed by a *0*
```
Comment (monitz87) [inline Readme.md]: Only

File: Readme.md [hunk 2]
```diff
@@ -1,71 +1,115 @@
...
+    t.date('12132012'); // Invalid
+    t.date('32/12/2013'); // Invalid, day out of range
 
 Get:
 
-    t.description();
+    t.date(); // Returns a string formatted as *dd/mm/yyyy* and prefixes day and month with a *0* when necessary
+---
+**Kind**
+Type must be a string and must be in the list of valid types (default is "normal")
```
Comment (monitz87) [inline Readme.md]: change type to kind here

File: Readme.md [hunk 3]
```diff
@@ -1,71 +1,115 @@
...
 
-**Charge and Deposit**
-Only positive numbers are allowed
 Set:
 
-    t.charge(10);
-    t.deposit(100);
+    t.type("normal"); // Valid
```
Comment (monitz87) [inline Readme.md]: same here

File: transaction.js [hunk 4]
```diff
@@ -1,324 +1,277 @@
...
+
+TRANSACTION_TYPES = ['normal', 'due_cash', 'due_commerce', 'due_fixed', 'due_advance',
+  'total_due'];
+VALID_KEYS = ['date', 'kind', 'balance', 'charge', 'deposit', 'description', 'extendedDescription',
+  'serial', 'dues'];
+
+// Private methods
+/**
+ * Validates that the input string is a valid date formatted as "mm/dd/yyyy"
+ * @param  {String}  dateString String with dare to check.
```
Comment (monitz87) [inline transaction.js]: the comment says 'dare' instead of 'date'

**Comment by monitz87** [comment]
Looks good :D

File: test/transaction.spec.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        t.date('01/06/2014');
+        t.kind('normal');
+        t.balance(0);
+        t.charge(15000);
+        t.deposit(0);
+        t.description('Giro cajero automatico');
+
+        t.build();
+      };
+      assert.doesNotThrow(fn, Error);
```
Comment (RafaelVidaurre) [inline test/transaction.spec.js]: This is not necessary as the test runner will fail the tests when an exception is thrown.

It should be:

``` javascript
  var t = new Transaction();

  t.date('01/06/2014');
  t.kind('normal');
  t.balance(0);
  t.charge(15000);
  t.deposit(0);
  t.description('Giro cajero automatico');
  t.build();
```

**Comment by RafaelVidaurre** [comment]
Fix tests and merge gogogo

**Pull request merged**